tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -694,12 +692,22 @@ def convert_pytorch_to_onnx(model, input_shape, opset_version, example_inputs, o
             inputs = to_torch_tensor(inputs)
     elif input_shape is not None:
         inputs = []
-        for shape_idx, shape in enumerate(input_shape):
+        for shape in input_shape:
             static_shape = get_static_shape(shape, dynamic_value=1)
             inputs.append(torch.zeros(static_shape))
         inputs = tuple(inputs)
     else:
-        raise Error("Please provide input_shape or example_input for converting PyTorch model.")
+        if not allow_none:
+            raise Error("Please provide input_shape or example_input for converting PyTorch model.")
+    return inputs
+
+
+def convert_pytorch_to_onnx(model, input_shape, opset_version, example_inputs, output_dir):
+    import io
+    import torch
+
```

Comment:
please put all PyTorch specific routines like `convert_pytorch_to_onnx` and  `get_pytorch_decoder` to the separate python file like `moc_frontend/pytorch_frontend_utils.py`. Now `convert_impl.py` becomes (or already became) too overloaded:)

Reply:
I belive we can have cyclic dependency if I move conversion via onnx, as is it uses _convert inside...